Code/Source/svFSI/eq_assem.cpp

```diff
@@ -302,6 +302,74 @@ void b_neu_folw_p(ComMod& com_mod, const bcType& lBc, const faceType& lFa, const
   }
 }
 
+/// @brief Update the surface integral involved in the coupled/resistance BC
+/// contribution to the stiffness matrix to reflect deformed geometry, if using
+/// a follower pressure load.
+/// The value of this integral is stored in lhs%face%val.
+/// This integral is sV = int_Gammat (Na * n_i) (See Brown et al. 2024, Eq. 56)
+/// where Na is the shape function and n_i is the normal vector.
+///
+/// This function updates the variable lhs%face%val with the new value, which
+/// is eventually used in ADDBCMUL() in the linear solver to add the contribution
+/// from the resistance BC to the matrix-vector product of the tangent matrix and
+/// an arbitrary vector.
+void fsi_ls_upd(ComMod& com_mod, const bcType& lBc, const faceType& lFa)
+{
+  using namespace consts;
+  using namespace utils;
+  using namespace fsi_linear_solver;
+
+  #define n_debug_fsi_ls_upd
+  #ifdef debug_fsi_ls_upd
+  DebugMsg dmsg(__func__, com_mod.cm.idcm());
+  dmsg.banner();
+  dmsg << "lFa.name: " << lFa.name;
+  #endif
+
+  auto& cm = com_mod.cm;
+  int nsd = com_mod.nsd;
+  int tnNo = com_mod.tnNo;
+
+  int iM = lFa.iM;
+  int nNo = lFa.nNo;
+
+  Array<double> sVl(nsd,nNo); 
+  Array<double> sV(nsd,tnNo); 
+
+  // Updating the value of the surface integral of the normal vector
+  // using the deformed configuration ('n' = new = timestep n+1)
+  sV = 0.0;
+  for (int e = 0; e < lFa.nEl; e++) {
+    if (lFa.eType == ElementType::NRB) {
+      // CALL NRBNNXB(msh(iM),lFa,e)
+    }
+    for (int g = 0; g < lFa.nG; g++) {
+      Vector<double> n(nsd);
+      auto Nx = lFa.Nx.slice(g);
```

Comment:
It is more efficient to use the **Array3::rslice()** method here, does not copy data.

Reply:
Done, thanks

Reply:
I also changed all other instances of slice() in this file to rslice(). Each instance uses slice() in the same way: get shape function Nx for the current Gauss point g, and pass it to gnnb(). Since Nx is treated as a constant in gnnb(), and Nx isn't used anywhere else, I think it's fine to switch all instances to rslice()